src/UnigramWordPredictor.java

```diff
@@ -99,10 +115,27 @@ public void train(Scanner scanner) {
    * @return the predicted next word, or null if no prediction can be made
    */
   public String predictNextWord(List<String> context) {
-    // TODO: Return a predicted word given the words preceding it
+    // DONE: Return a predicted word given the words preceding it
     // Hint: only the last word in context should be looked at
-    return null;
-  }
+    if (context == null || context.isEmpty()) { // check if null or empty
+      return null;
+    }
+
+    // getting the last word from context
+    String lastWord = context.get(context.size() - 1);
+
+    // retrieving next possible words from neighborMap
+    List<String> possibleNextWords = neighborMap.get(lastWord);
+
+    // if there are no next words available
+    if (possibleNextWords == null || possibleNextWords.isEmpty()) {
+      return null;
+    }
```

Comment:
Nice thinking through of the possibilities here. The assignment didn't specify what to do if we see a word that we've never seen a following word to. Returning `null` is one possible way to handle it, what are other possibilities you can think of?

Reply:
Thank you for your concise feedback! Instead of returning `null`, I could have returned a different statement. 

Possible Solutions:
1.  Return an **empty** list -> indicate there are no possible next words left.
2. Return an **empty** string -> `return " ";` 
3. Throw an _exception_.